Code/Tools/ProjectManager/Source/ProjectButtonWidget.cpp

```diff
@@ -267,47 +325,151 @@ namespace O3DE::ProjectManager
+        m_projectImageLabel->GetOpenEditorButton()->hide();
+
+        ShowMessage(tr("Opening Editor..."));
+    }
+
+    void ProjectButton::ShowBuildRequiredState()
+    {
+        ShowBuildButton();
+
+        SetProjectBuilding(false);
+
+        ShowWarning(tr("Project build required."));
+    }
+
+    void ProjectButton::ShowBuildingState()
+    {
+        m_projectImageLabel->GetShowLogsButton()->show();
+
+        // Setting project to building also disables launching
+        SetProjectBuilding(true);
+
+        ShowMessage(tr("Building Project..."));
+    }
+
+    void ProjectButton::ShowBuildFailedState()
+    {
+        ShowBuildButton();
+
+        SetProjectBuilding(false);
+
+        // Show, show logs button if avaliable
+        m_projectImageLabel->GetShowLogsButton()->setVisible(!m_logUrl.isEmpty());
+
+        ShowWarning(tr("Failed to build"));
     }
 
     void ProjectButton::SetProjectButtonAction(const QString& text, AZStd::function<void()> lambda)
     {
-        QPushButton* projectActionButton = m_projectImageLabel->GetActionButton();
-        if (!m_actionButtonConnection)
+        QPushButton* projectActionButton;
+        QPushButton* projectOtherActionButton;
+
+        if (text.contains("Cancel", Qt::CaseInsensitive))
         {
-            projectActionButton->setVisible(true);
+            // Use red button is action involves cancelling
+            projectActionButton = m_projectImageLabel->GetActionCancelButton();
+            projectOtherActionButton = m_projectImageLabel->GetActionButton();
         }
         else
+        {
+            projectActionButton = m_projectImageLabel->GetActionButton();
+            projectOtherActionButton = m_projectImageLabel->GetActionCancelButton();
+        }
+
+        if (m_actionButtonConnection)
         {
             disconnect(m_actionButtonConnection);
         }
 
+        projectActionButton->setVisible(true);
+        projectOtherActionButton->setVisible(false);
+
         projectActionButton->setText(text);
         projectActionButton->setMenu(nullptr);
         m_actionButtonConnection = connect(projectActionButton, &QPushButton::clicked, lambda);
     }
 
-    void ProjectButton::ShowDefaultBuildButton()
+    void ProjectButton::SetBuildLogsLink(const QUrl& logUrl)
+    {
+        m_logUrl = logUrl;
+    }
+
+    void ProjectButton::SetContextualText(const QString& text)
+    {
+        if (m_currentState == ProjectState::Building)
+        {
+            // Don't update for empty build progress messages
+            if (!text.isEmpty())
+            {
+                // Show info about what's currently building
+                ShowMessage({}, text);
+            }
+        }
+        else
+        {
+            ShowMessage(text);
+        }
+    }
+
+    void ProjectButton::ShowBuildButton()
     {
         QPushButton* projectActionButton = m_projectImageLabel->GetActionButton();
+        projectActionButton->setObjectName("projectBuildButton");
```

Comment:
This object name (projectBuildButton) does not appear to be used in the qss file

Reply:
Nice, catch forgot to remove that.